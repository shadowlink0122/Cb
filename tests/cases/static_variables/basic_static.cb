// 基本的なstatic変数の動作テスト
int counter(){
    static int count = 0;
    count++;
    return count;
}

int main(){
    // static変数は初期化されて、値が保持される
    int result1 = counter(); // 1
    int result2 = counter(); // 2
    int result3 = counter(); // 3
    int result4 = counter(); // 4
    int result5 = counter(); // 5
    
    // 値の検証
    if (result1 != 1) {
        println("ERROR: First counter() should be 1, got %d", result1);
        return 1;
    }
    if (result2 != 2) {
        println("ERROR: Second counter() should be 2, got %d", result2);
        return 1;
    }
    if (result3 != 3) {
        println("ERROR: Third counter() should be 3, got %d", result3);
        return 1;
    }
    if (result4 != 4) {
        println("ERROR: Fourth counter() should be 4, got %d", result4);
        return 1;
    }
    if (result5 != 5) {
        println("ERROR: Fifth counter() should be 5, got %d", result5);
        return 1;
    }
    
    println("%d", result1); // 1
    println("%d", result2); // 2
    println("%d", result3); // 3
    println("%d", result4); // 4
    println("%d", result5); // 5
    println("Basic static test passed");
    return 0;
}
