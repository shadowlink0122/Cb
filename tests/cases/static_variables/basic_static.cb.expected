1
2
3
4
5
Basic static test passed
